smaily_for_drupal/src/Form/SubscribeForm.php

```diff
@@ -76,6 +78,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     // Remove token so as to not send it to Smaily as a field.
     $form['#token'] = FALSE;
     $form['#after_build'][] = [$this, 'removeHiddenDrupalInputs'];
+    $this->handleResponseMessage();
```

Comment:
This will be executed only when you have a form on your page.

Reply:
This is correct. Messages are shown when you are redirected back, otherwise the success/failure page will give the user feedback. I can't imagine how to create a site-wide listener and wouldn't it be overkill for this module?

Reply:
Maybe you can use [hook_preprocess](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Render%21theme.api.php/function/hook_preprocess/8.2.x) to show success/failure page response message?

Reply:
I think hook_preprocess is used for themes only. I would need a .theme file for that.

Reply:
`hook_preprocess` works with modules too. `smaily_for_drupal_preprocess` would be executed for each visible block. But I guess `hook_page_attachments` works too.